File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/TypeTemplateService.java
package com.pinyougou;
import java.util.List;
import java.util.Map;

import com.github.pagehelper.PageInfo;

import com.pinyougou.pojo.TypeTemplate;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface TypeTemplateService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TypeTemplate> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo<TypeTemplate> findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TypeTemplate typeTemplate);
	
	
	/**
	 * 修改
	 */
	public void update(TypeTemplate typeTemplate);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TypeTemplate findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo<TypeTemplate> findPage(TypeTemplate typeTemplate, int pageNum, int pageSize);

	/**
	 * 由于在模板表中只存储了规格名称，没有规格值，所以新增方法查询规格列表（规格名+规格值）
	 * 返回规格列表
	 * @param id
	 * @return
	 */
	public List<Map> findSpecList(Long id);
	
}

File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/typeTemplateService.js
//服务层
app.service('typeTemplateService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/typeTemplate/findAll.shtml');
	}
	//分页 
	this.findPage=function(page,size){
		return $http.get('/typeTemplate/findPage.shtml?page='+page+'&size='+size);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/typeTemplate/findOne.shtml?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/typeTemplate/add.shtml',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/typeTemplate/update.shtml',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/typeTemplate/delete.shtml?ids='+ids);
	}
	//搜索
	this.search=function(page,size,searchEntity){
		return $http.post('/typeTemplate/search.shtml?page='+page+"&size="+size, searchEntity);
	}

	//根据规格id查找规格选项列表
	this.findSpecList=function (id) {
		return $http.get("/typeTemplate/findSpecList.shtml?id=" + id);
	}
});
